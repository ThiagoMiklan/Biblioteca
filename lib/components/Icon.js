"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

require("../css/all");

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string",
  right: "boolean",
  onClick: "function",
  icon_name: "string"
};

var Icon = function Icon(props) {
  (0, _type_validations.validate)(props, props_obj, "Icon");
  var definition = props.definition == undefined ? "icon" : "icon " + props.definition;
  return /*#__PURE__*/React.createElement(React.Fragment, null, props.right == false || props.right == undefined ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", {
    onClick: props.onClick,
    className: definition
  }, /*#__PURE__*/React.createElement("i", {
    className: props.icon_name
  })), /*#__PURE__*/React.createElement("span", null, props.children)) : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", null, props.children), /*#__PURE__*/React.createElement("span", {
    onClick: props.onClick,
    className: definition
  }, /*#__PURE__*/React.createElement("i", {
    className: props.icon_name
  }))));
};

var _default = Icon;
exports.default = _default;