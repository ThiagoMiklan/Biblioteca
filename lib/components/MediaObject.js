"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _Image = _interopRequireDefault(require("./Image"));

var _Content = _interopRequireDefault(require("./Content"));

var _Button = _interopRequireDefault(require("./Button"));

var _Level = _interopRequireDefault(require("./Level"));

var _type_validations = require("../tools/type_validations");

var _props_obj;

var props_obj = (_props_obj = {
  definition: 'string',
  image_definition: 'string',
  figure_definition: 'string',
  content_definition: 'string',
  src: 'string',
  itens: 'array',
  level_definition: 'string',
  left_definition: 'string',
  right_definition: 'string'
}, (0, _defineProperty2.default)(_props_obj, "content_definition", 'string'), (0, _defineProperty2.default)(_props_obj, "onClickDelete", 'function'), _props_obj);

var MediaObject = function MediaObject(props) {
  var definition = props.definition == undefined ? "media" : "media " + props.definition;
  (0, _type_validations.validate)(props, props_obj, "MediaObject");
  return /*#__PURE__*/React.createElement("article", {
    className: definition
  }, assembleLeftMedia(props), assembleContentMedia(props), assembleRightMedia(props));
};

function assembleLeftMedia(props) {
  var definition = props.left_definition == undefined ? "media-left" : "media-left " + props.left_definition;
  return /*#__PURE__*/React.createElement("figure", {
    className: definition
  }, /*#__PURE__*/React.createElement(_Image.default, {
    definition: props.figure_definition,
    image_definition: props.image_definition,
    src: props.src
  }));
}

function assembleContentMedia(props) {
  var definition = props.content_definition == undefined ? "media-content" : "media-content " + props.content_definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, /*#__PURE__*/React.createElement(_Content.default, {
    definition: definition
  }, props.children), assembleDefaultLevel(props));
}

function assembleDefaultLevel(props) {
  return /*#__PURE__*/React.createElement(_Level.default, {
    definition: props.level_definition,
    itens: props.itens
  });
}

function assembleRightMedia(props) {
  var definition = props.right_definition == undefined ? "media-right" : "media-right " + props.right_definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.right != undefined ? props.right : /*#__PURE__*/React.createElement(_Button.default, {
    onClick: props.onClickDelete,
    delete: true
  }, " "));
}

var _default = MediaObject;
exports.default = _default;