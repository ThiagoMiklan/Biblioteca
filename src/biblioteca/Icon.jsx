// @flow
import * as React from 'react'
import classnames from 'classnames';
import getClassNames from '../tools/getClassName';
import PropTypes from 'prop-types';
import '../css/all';
import checkerDefinition from '../tools/checker.js';



type Props = {
    definition?: string,
    icon_right?: bool,
    onClick? : ()=> void,
    children?: React.Node,
    icon_name?: string
}

const Icon = (props:Props) => {
    checkerDefinition(props.definition,"Icon");
    var icon_definition  = (props.definition == undefined) ? "": props.definition;
    var definition =  classnames(getClassNames("icon "+icon_definition, "Icon"));
    return (
        <>  
            {(props.icon_right == false || props.icon_right == undefined) && <span>{props.children}</span>}
            <span onClick={props.onClick} className={definition}>
                <i className={props.icon_name} aria-hidden="true" ></i>
            </span>
            {props.icon_right == true && <span>{props.children}</span>}
            
        </>
    );
}

function Teste(){
    var x = <Icon ic></Icon>
}




export default Icon;