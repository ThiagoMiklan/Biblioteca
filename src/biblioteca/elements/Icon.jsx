import React from 'react'

// ProvisÃ³rio
const Icon = (props) => {
    return (
        <span class={props.definition}>
            <i class={props.icon}></i>
        </span>
    );
}

export default Icon;