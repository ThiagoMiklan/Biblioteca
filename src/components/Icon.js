// @flow
import * as React from 'react'
import { validate } from '../tools/type_validations.js';

type Props = {
    definition?: string,
    right?: bool,
    onClick?: () => void,
    children?: React.Node,
    icon_name?: string
}

const props_obj = {
    definition: "string",
    right: "boolean",
    onClick: "function",
    icon_name: "string"
}

const Icon = (props: Props) => {
    validate(props, props_obj,"Icon");
    var definition = (props.definition == undefined) ? "icon" : "icon " + props.definition;
    return (
        <>
            {(props.right == false || props.right == undefined) ?
                <>
                    <span onClick={props.onClick} className={definition}>
                        <i className={props.icon_name}></i>
                    </span>
                    <span>{props.children}</span>
                </>
                :
                <>
                    <span>{props.children}</span>
                    <span onClick={props.onClick} className={definition}>
                        <i className={props.icon_name}></i>
                    </span>
                </>

            }

        </>
    );
}

export default Icon;

