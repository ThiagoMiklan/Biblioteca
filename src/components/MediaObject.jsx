// @flow
import * as React from 'react';
import Image from './Image';
import Content from './Content';
import Button from './Button';
import Level from './Level';

type Props = {
     definition?:string,
    image_definition?: string,
    figure_definition?: string,
    content_definition?: string,
    children?: React.Node,
    src?:string,
    right_element?: React.Node,
    onClickDelete?: ()=> void

}

const MediaObject = (props:Props)=>{
    return  <article class="media">
                {assembleLeftMedia(props)}
                {assembleContentMedia(props)}
                {assembleRightMedia(props)}
            </article>
}

function assembleLeftMedia(props: Props){
    return  <figure class="media-left">
        <Image definition={props.figure_definition} image_definition={props.image_definition} src={props.src}/>
  </figure>
}

function assembleContentMedia(props: Props){
    return <div class="media-content">
                <Content definition={props.content_definition}>
                     {props.children}
                </Content>
                {assembleDefaultLevel(props)}
            </div>
}   

function assembleDefaultLevel(props: Props){
    return <Level itens={props.itens}> 
    
            </Level>
}

function assembleRightMedia(props: Props){
    return <div class="media-right">
                {props.right_element != undefined ? props.right_element : <Button onClick={props.onClickDelete} delete={true}> </Button>
                }
            </div>
}


export default MediaObject;