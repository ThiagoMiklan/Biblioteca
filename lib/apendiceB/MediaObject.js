/*
import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import 'bulma/css/bulma.css'
import MediaObject from './components/MediaObject';
import Icon from './components/Icon';
import Level from './components/Level';
import Columns from './components/Columns';
import Column from './components/Column'

const itens = [
  {value:<Icon definition="is-small has-text-info" icon_name="fas fa-reply"></Icon>, left: true, onClick:()=>{alert("Clicked")}, key: 1},
  {value:<Icon definition="is-small has-text-info" icon_name="fas fa-retweet"></Icon>, left: true,key: 2},
  {value:<Icon definition="is-small has-text-info"  icon_name="fas fa-heart"></Icon>,left: true, key: 3}
]


ReactDOM.render(
<article class="media">
  <figure class="media-left">
    <p class="image is-64x64">
      <img src="https://bulma.io/images/placeholders/128x128.png"/>
    </p>
  </figure>
  <div class="media-content">
    <div class="content">
      <p>
        <strong>John Smith</strong> <small>@johnsmith</small> <small>31m</small>
        <br/>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin ornare magna eros, eu pellentesque tortor vestibulum ut. Maecenas non massa sem. Etiam finibus odio quis feugiat facilisis.
      </p>
    </div>
    <nav class="level is-mobile">
      <div class="level-left">
        <a class="level-item">
          <span class="icon is-small"><i class="fas fa-reply"></i></span>
        </a>
        <a class="level-item">
          <span class="icon is-small"><i class="fas fa-retweet"></i></span>
        </a>
        <a class="level-item">
          <span class="icon is-small"><i class="fas fa-heart"></i></span>
        </a>
      </div>
    </nav>
  </div>
  <div class="media-right">
    <button class="delete"></button>
  </div>
</article>
,
  document.getElementById("root"))

*/

/*
import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import 'bulma/css/bulma.css'
import MediaObject from './components/MediaObject';
import Icon from './components/Icon';
import Level from './components/Level';
import Columns from './components/Columns';
import Column from './components/Column'

const itens = [
  {value:<Icon definition="is-small has-text-info" icon_name="fas fa-reply"></Icon>, left: true, onClick:()=>{alert("Clicked")}, key: 1},
  {value:<Icon definition="is-small has-text-info" icon_name="fas fa-retweet"></Icon>, left: true,key: 2},
  {value:<Icon definition="is-small has-text-info"  icon_name="fas fa-heart"></Icon>,left: true, key: 3}
]


ReactDOM.render(
  <Columns definition="is-multiline">
    <Column definition="is-half is-spaced">
        <MediaObject itens={itens} src="https://bulma.io/images/placeholders/128x128.png" image_definition="is-64x64" level_definition="is-mobile"> 
          <p>
            <strong>John Smith</strong> <small>@johnsmith</small> <small>31m</small>
            <br/>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin ornare magna eros, eu pellentesque tortor vestibulum ut. Maecenas non massa sem. Etiam finibus odio quis feugiat facilisis.
          </p>
      </MediaObject> 
    </Column>
    
  </Columns>
,
  document.getElementById("root"))

*/
"use strict";