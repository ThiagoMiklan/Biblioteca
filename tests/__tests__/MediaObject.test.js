import React from 'react';
import renderer from 'react-test-renderer';
import MediaObject from '../../src/components/MediaObject';
import Icon from '../../src/components/Icon';


const itens = [
  {value:<Icon definition="is-small has-text-info" icon_name="fas fa-reply"></Icon>, left: true, onClick:()=>{alert("Clicked")}, key: 1},
  {value:<Icon definition="is-small has-text-info" icon_name="fas fa-retweet"></Icon>, left: true,key: 2},
  {value:<Icon definition="is-small has-text-info"  icon_name="fas fa-heart"></Icon>,left: true, key: 3}
]


test('images normal',()=>{
    const component = renderer.create(
       <MediaObject itens={itens} src="https://bulma.io/images/placeholders/128x128.png" image_definition="is-64x64" level_definition="is-mobile"> 
          <p>
            <strong>John Smith</strong> <small>@johnsmith</small> <small>31m</small>
            <br/>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin ornare magna eros, eu pellentesque tortor vestibulum ut. Maecenas non massa 
            sem. Etiam finibus odio quis feugiat facilisis.
          </p>
      </MediaObject> 
  ).toJSON();
    expect(component).toMatchSnapshot();
});

