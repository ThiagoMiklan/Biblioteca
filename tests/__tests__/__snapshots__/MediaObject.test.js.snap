// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`images normal 1`] = `
<article
  className="media"
>
  <figure
    className="media-left"
  >
    <figure
      className="image"
    >
      <img
        className="is-64x64"
        src="https://bulma.io/images/placeholders/128x128.png"
      />
    </figure>
  </figure>
  <div
    className="media-content"
  >
    <div
      className="content media-content"
    >
      <p>
        <strong>
          John Smith
        </strong>
         
        <small>
          @johnsmith
        </small>
         
        <small>
          31m
        </small>
        <br />
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin ornare magna eros, eu pellentesque tortor vestibulum ut. Maecenas non massa sem. Etiam finibus odio quis feugiat facilisis.
      </p>
    </div>
    <nav
      className="level is-mobile"
    >
      <div
        className="level-left"
      >
        <p
          className="level-item"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            className="icon is-small has-text-info"
          >
            <i
              className="fas fa-reply"
            />
          </span>
          <span />
        </p>
        <p
          className="level-item"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            className="icon is-small has-text-info"
          >
            <i
              className="fas fa-retweet"
            />
          </span>
          <span />
        </p>
        <p
          className="level-item"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            className="icon is-small has-text-info"
          >
            <i
              className="fas fa-heart"
            />
          </span>
          <span />
        </p>
      </div>
      <div
        className="level-right"
      />
    </nav>
  </div>
  <div
    className="media-right"
  >
    <button
      className="delete"
      disabled={false}
    />
  </div>
</article>
`;
